tiny_dnn/layers/layer.h

```diff
@@ -310,6 +313,9 @@ class layer : public node {
    *important
    **/
   virtual serial_size_t fan_out_size() const { return out_shape()[0].width_; }
+  virtual serial_size_t fan_out_size(serial_size_t) const {
+    return fan_out_size();
+  }
```

Comment:
To initialize reurrent weights, I had to modify the fan in and fan out, I also modified it in the layer.h.